bl/blocks/block_factory/inc/block_factory.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <memory>
+
+#include "block.hpp"
+
+enum class BlockType { DIAGONAL, LSHAPE, TSHAPE, SQUARE };
+
+class BlockFactory {
+   public:
+    BlockFactory() = delete;
+    static std::unique_ptr<Block> createBlock(BlockType type);
```

Comment:
I'm pretty sure you'll change it to shared pointer later on, but if you want to keep it I'm ok with that :P

Reply:
I'll change it shared_ptr later on, but conversion from unique_ptr is possible while other way around it's not, so it's more versatile this way :D